File: common/src/main/java/mconi/common/sim/subsystem/SimulationContext.java
package mconi.common.sim.subsystem;

import mconi.common.sim.OniSimulationConfig;
import mconi.common.sim.OniSimulationGrid;

public class SimulationContext
{
	private final long serverTick;
	private final OniSimulationConfig config;
	private final OniSimulationGrid grid;

	public SimulationContext(long serverTick, OniSimulationConfig config, OniSimulationGrid grid)
	{
		this.serverTick = serverTick;
		this.config = config;
		this.grid = grid;
	}

	public long serverTick() { return serverTick; }
	public OniSimulationConfig config() { return config; }
	public OniSimulationGrid grid() { return grid; }
}

File: common/src/main/kotlin/mconi/common/sim/subsystem/SimulationContext.kt
package mconi.common.sim.subsystem

import mconi.common.sim.OniSimulationConfig
import mconi.common.sim.OniSimulationRuntime
import net.minecraft.server.level.ServerLevel

class SimulationContext(
    private val serverTick: Long,
    private val config: OniSimulationConfig,
    private val level: ServerLevel,
    private val runtime: OniSimulationRuntime,
) {
    fun serverTick(): Long = serverTick
    fun config(): OniSimulationConfig = config
    fun level(): ServerLevel = level
    fun runtime(): OniSimulationRuntime = runtime
}
